### PR TITLE
feat(option/internaloption): add WithDefaultEndpointTemplate

### PR DESCRIPTION
**Pull request by quartzmo**
* Add DefaultEndpointTemplate to internal/settings.go
* Deprecate internaloption.WithDefaultEndpoint

refs: #2264

### CODE REVIEW

File: option/internaloption/internaloption.go [hunk 1]
```diff
@@ -22,10 +22,29 @@ func (o defaultEndpointOption) Apply(settings *internal.DialSettings) {
 // It should only be used internally by generated clients.
 //
 // This is similar to WithEndpoint, but allows us to determine whether the user has overridden the default endpoint.
+//
+// Deprecated: WithDefaultEndpoint does not support setting the universe domain.
```
Comment (codyoss) [inline option/internaloption/internaloption.go:26]: Any reason to not just repurpose this option. I think we could since it is internal?

Comment (quartzmo) [reply option/internaloption/internaloption.go:26]: I actually plan to use both options simultaneously during a transitional period as universe domain support is incrementally expanded.

Comment (codyoss) [reply option/internaloption/internaloption.go:26]: could you do the same thing with one and just check for a %s? I think either way is fine as long as we plan to fully remove the old one eventually

**Review by codyoss — APPROVED** [review]

File: option/internaloption/internaloption.go (hunk 1 above)
Comment (quartzmo) [reply option/internaloption/internaloption.go:26]: Definitely plan to remove the old one.

**Pull request merged**